sdk/storage/ustorageclient/CMakeLists.txt

```diff
@@ -0,0 +1,71 @@
+#Copyright (c) Microsoft. All rights reserved.
+#Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+cmake_minimum_required(VERSION 3.2)
+
+project(azure_iot_ustorageclient)
+
+set(AZURE_IOT_USTORAGECLIENT_VERSION 1.0.1)
+
+#Include helper functions
+include("build_all/configs/ustorageclient_functions.cmake")
+include("build_all/configs/ustorageclient_build_rules.cmake")
+
+#Set the standards for C and C++
+add_standards()
+
+#Specify user options
+option(run_azstorage_e2e_tests "set run_e2e_tests to ON to run e2e tests (default is OFF)" OFF)
+option(run_azstorage_unit_tests "set run_unittests to ON to run unit tests (default is OFF)" OFF)
+option(skip_azstorage_samples "set skip_samples to ON to skip building samples (default is OFF)[if possible, they are always built]" OFF)
+option(use_azstorage_installed_dependencies "set use_installed_dependencies to ON to use installed packages instead of building dependencies from submodules" OFF)
+
+#Set ustorageclient c files
+set(azure_iot_ustorageclient_c_files
+    ${PROJECT_SOURCE_DIR}/src/azstorage_blob.c
+)
+
+#Add library of ustorageclient c files
+add_library(azure_iot_ustorageclient ${azure_iot_ustorageclient_c_files})
+
+#Set the include directories for this project
+set(azure_iot_ustorageclient_h_dir
+    ${PROJECT_SOURCE_DIR}/inc
+    ${PROJECT_SOURCE_DIR}/inc/_internal
+    ${PROJECT_SOURCE_DIR}/config
+)
+
+#Add include directories for this target and anyone linking against it
+target_include_directories(azure_iot_ustorageclient PUBLIC ${azure_iot_ustorageclient_h_dir})
+target_include_directories(azure_iot_ustorageclient PRIVATE
```

Comment:
prefer using (possibly) imported targets and target_link_libraries instead of target_include_directories for dependency include directories. (if targets don't correctly set their include directories then fix them "upstream".